paymentsheet/src/androidTest/java/com/stripe/android/paymentsheet/RetryRule.kt

```diff
@@ -0,0 +1,32 @@
+package com.stripe.android.paymentsheet
+
+import android.util.Log
+import org.junit.AssumptionViolatedException
+import org.junit.rules.TestRule
+import org.junit.runner.Description
+import org.junit.runners.model.Statement
+
+internal class RetryRule(private val attempts: Int) : TestRule {
+    override fun apply(base: Statement, description: Description): Statement {
+        val logTag = description.className
+        return object : Statement() {
+            @Throws
+            override fun evaluate() {
+                for (attempt in 1..attempts) {
+                    val isLast = attempts == attempt
+                    val error = runCatching { base.evaluate() }.exceptionOrNull()
+
+                    if (error != null) {
```

Comment:
can you also write this as:
```
runCatching {
  base.evaluate() 
}.onSuccess {
  return
}.onFailure {
...
} 
```

That might be more readable/save you the "// The test succeeded" comment